File: test/entity/token.ts
import { Entity, Id, Column, ManyToOne, JoinColumn, OneToMany } from '../../core/decorator/decorator'
import { BaseEntity } from '../../core/baseentity'
import { EFkConstraint } from '../../core/entitydefine'
import {User} from './user'

@Entity("t_token",'tunnel')
export class Token extends BaseEntity{
	@Id()
	@ManyToOne({entity:'User',eager:false})
	@JoinColumn({name:'user_id',refName:'user_id'})
	private user:User;

	@Column({
		name:'token',
		type:'string',
		nullable:true
	})
	private token:string;

	@Column({
		name:'create_time',
		type:'int',
		nullable:true
	})
	private createTime:number;

	@Column({
		name:'disable_time',
		type:'int',
		nullable:true
	})
	private disableTime:number;

	public getUser():User{
		return this.user;
	}
	public setUser(value:User){
		this.user = value;
	}

	public getToken():string{
		return this.token;
	}
	public setToken(value:string){
		this.token = value;
	}

	public getCreateTime():number{
		return this.createTime;
	}
	public setCreateTime(value:number){
		this.createTime = value;
	}

	public getDisableTime():number{
		return this.disableTime;
	}
	public setDisableTime(value:number){
		this.disableTime = value;
	}

}

File: jpa/pojo/Token.java
package dao.pojo;

// default package
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.PrimaryKeyJoinColumn;
import javax.persistence.Table;
import javax.persistence.TableGenerator;

/**
 * Token entity. @author MyEclipse Persistence Tools
 */
@Entity
@Table(name = "t_token", catalog = "tiandian")
public class Token extends BasePojo implements java.io.Serializable {
	// Fields
	private Long user_id;
	private User user;
	private String token;
	private Long create_time;
	private Long disable_time;

	// Constructors
	/** default constructor */
	public Token() {
	}

	/** minimal constructor */
	public Token(Long token_id, Long create_time, Long disable_time) {
		this.user_id = token_id;
		this.create_time = create_time;
		this.disable_time = disable_time;
	}

	/** full constructor */
	public Token(Long token_id, User user, String token, Long create_time, Long disable_time) {
		this.user_id = token_id;
		this.user = user;
		this.token = token;
		this.create_time = create_time;
		this.disable_time = disable_time;
	}

	// Property accessors
	@Id
	@Column(name = "user_id", unique = true, nullable = false)
	public Long getUser_id() {
		return this.user_id;
	}

	public void setUser_id(Long token_id) {
		this.user_id = token_id;
	}

	@OneToOne(fetch = FetchType.LAZY)
	@PrimaryKeyJoinColumn
	public User getUser() {
		return this.user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	@Column(name = "token", length = 32)
	public String getToken() {
		return this.token;
	}

	public void setToken(String token) {
		this.token = token;
	}

	@Column(name = "create_time", nullable = false, length = 19)
	public Long getCreate_time() {
		return this.create_time;
	}

	public void setCreate_time(Long create_time) {
		this.create_time = create_time;
	}

	@Column(name = "disable_time", nullable = false, length = 19)
	public Long getDisable_time() {
		return this.disable_time;
	}

	public void setDisable_time(Long disable_time) {
		this.disable_time = disable_time;
	}

	@Override
	public Object getEntityId() {
		return this.user_id;
	}
}